src/vs/workbench/browser/parts/views/treeView.ts

```diff
@@ -755,10 +755,19 @@ abstract class AbstractTreeView extends Disposable implements ITreeView {
 		const customTreeKey = RawCustomTreeViewContextKey.bindTo(this.tree.contextKeyService);
 		customTreeKey.set(true);
 		this._register(this.tree.onContextMenu(e => this.onContextMenu(treeMenus, e, actionRunner)));
-		this._register(this.tree.onDidChangeSelection(e => this._onDidChangeSelection.fire(e.elements)));
+
+		this._register(this.tree.onDidChangeSelection(e => {
+			if (e.elements !== this.lastSelection) {
```

Comment:
It seems `abstractTree.ts` already filters along with `extHostTreeView.ts`

Is this really needed? Especially that those arrays are compared by reference